microgrid_optimizer.py

```diff
@@ -1443,52 +1466,96 @@ def save_results_to_file(self, spg, filename='simulation_results'):
-             'pv_area_ft2': 'PV Area', 'capital_cost_usd': 'Capital Cost',
-             'pv_capital': 'PV Capital', 'battery_capital': 'Battery Capital',
-             'generator_capital': 'Generator Capital',
-             'fuel_tank_capital': 'Fuel Tank Capital',
-             'pv_o&m': 'PV O&M', 'battery_o&m': 'Battery O&M',
-             'generator_o&m': 'Generator O&M',
-             'annual_benefits_usd': 'Annual PV Net-meter Revenue',
-             'demand_benefits_usd': 'Annual PV Demand Savings',
-             'simple_payback_yr': 'Simple Payback',
-             'pv_percent mean': 'PV Percent',
-             'batt_percent mean': 'Battery Percent',
-             'gen_percent mean': 'Generator Percent',
-             'fuel_used_gal mean': 'Fuel used (average scenario)',
-             'fuel_used_gal most-conservative': 'Fuel used (conservative scenario)'},
-                              inplace=True)
-
-        # Add units
-        units = ['kW', 'kWh', 'kW', 'kW', 'kW', 'gallons', 'ft^2', '$', '$',
-                 '$', '$', '$', '$/year', '$/year', '$/year', '$/year',
-                 '$/year', 'years', '%', '%', '%', 'gallons', 'gallons']
-        for _ in self.gen_power_percent:
-            units += ['kW', 'gallons', 'gallons', '$', '', '', '', 'kWh',
-                      'kWh', 'kW', 'kW']
+        if not self.existing_generator:
+            # Re-order columns
+            format_results = format_results[
+                ['pv_capacity', 'battery_capacity', 'battery_power',
+                 'generator_power_kW mean',
+                 'generator_power_kW most-conservative', 'fuel_tank_size_gal',
+                 'pv_area_ft2', 'capital_cost_usd',
+                 'pv_capital', 'battery_capital', 'generator_capital',
+                 'fuel_tank_capital', 'pv_o&m', 'battery_o&m', 'generator_o&m',
+                 'annual_benefits_usd', 'demand_benefits_usd',
+                 'simple_payback_yr', 'pv_percent mean', 'batt_percent mean',
+                 'gen_percent mean', 'fuel_used_gal mean',
+                 'fuel_used_gal most-conservative'] + list(
+                    format_results.columns[26:])]
+        else:
+            # Re-order columns
+            format_results = format_results[
+                ['pv_capacity', 'battery_capacity', 'battery_power', 'fuel_tank_size_gal',
+                 'pv_area_ft2', 'capital_cost_usd',
+                 'pv_capital', 'battery_capital', 'generator_capital',
+                 'fuel_tank_capital', 'pv_o&m', 'battery_o&m', 'generator_o&m',
+                 'annual_benefits_usd', 'demand_benefits_usd',
+                 'simple_payback_yr', 'pv_percent mean', 'batt_percent mean',
+                 'gen_percent mean', 'fuel_used_gal mean',
+                 'fuel_used_gal most-conservative'] + list(
+                    format_results.columns[26:])]
+
+        if not self.existing_generator:
+            # Rename columns
+            format_results.rename(columns=
```

Comment:
I believe you can use the same code for both modes and it will just ignore any columns that aren't in the dataframe. You'll just have to add a line for "generator_power".